app/src/main/java/com/github/se/icebreakrr/model/ai/AiViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package com.github.se.icebreakrr.model.ai
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.github.se.icebreakrr.model.profile.Profile
+import com.github.se.icebreakrr.model.profile.ProfilesViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+class AiViewModel(
+    private val aiRepository: AiRepository,
+    private val profilesViewModel: ProfilesViewModel
+) : ViewModel() {
+
+  sealed class UiState {
+    object Idle : UiState() // Initial state
+
+    object Loading : UiState() // API call in progress
+
+    data class Success(val data: String) : UiState() // Successful API response
+
+    data class Error(val message: String) : UiState() // Error occurred
+  }
+
+  private val _uiState = MutableStateFlow<UiState>(UiState.Idle)
+  val uiState: StateFlow<UiState>
+    get() = _uiState
+
+  private val systemPrompt =
```

Comment:
I tried prompt engineering a bit to make this prompt better, I suggest the following small improvement:

You are an expert in social interaction. We have an app that incentivises people cold approaching other people. Your goal is to help a user approaching another one by generating a cold approach phrase that they can use. I will provide metadata for the two users: "selfProfile" and "selectedProfile." Based on their profiles, generate a fun, engaging, and friendly icebreaker sentence that selfProfile can use to start a conversation with selectedProfile. You can use the interests to personalise them a bit but dont overfocus on them, the phrase has to be general enough and sets a positive tone for a friendly chat. AVOID AT ALL COST BEING CRINGE. Here are the details of two profiles:

Ideally we should also add examples of good catch phrases because it often makes the prompt exponentially better